services/git/commit.go

```diff
@@ -42,9 +43,19 @@ func ParseCommitsWithSignature(ctx context.Context, oldCommits []*user_model.Use
 			}
 		}
 
+		key := committer.Email
+		if c.Signature != nil {
+			key += fmt.Sprintf("-%s", c.Signature.Signature)
+		}
+		verification, ok := cachedVerifications[key]
+		if !ok {
+			verification = asymkey_service.ParseCommitWithSignatureCommitter(ctx, c.Commit, committer)
+			cachedVerifications[key] = verification
+		}
```

Comment:
How could it be right?

Reply:
In the `ParseCommitWithSignatureCommitter`, the returned result will only be related to `commit.Signature` and `email`.

Reply:
`ParseCommitWithSignatureCommitter` returns `*asymkey_model.CommitVerification`

```
type CommitVerification struct {
	Verified       bool
	Warning        bool
	Reason         string
	SigningUser    *user_model.User
	CommittingUser *user_model.User
	SigningEmail   string
	SigningKey     *GPGKey
	SigningSSHKey  *PublicKey
	TrustStatus    string
}
```


Then this pointer is reused for different commits?